clerktest/clerktest.go

```diff
@@ -34,11 +37,14 @@ func (rt *RoundTripper) RoundTrip(r *http.Request) (*http.Response, error) {
 	if rt.Status == 0 {
 		rt.Status = http.StatusOK
 	}
+	if rt.Method != "" {
+		require.Equal(rt.T, rt.Method, r.Method)
+	}
 	if rt.Path != "" {
 		require.Equal(rt.T, rt.Path, r.URL.Path)
 	}
-	if rt.Method != "" {
-		require.Equal(rt.T, rt.Method, r.Method)
+	if rt.Query != nil {
+		require.Equal(rt.T, rt.Query.Encode(), r.URL.Query().Encode())
```

Comment:
❓ Is order deterministic when encoding multiple parameters? It might be, just pointing out in case it causes any flakiness whatsoever. I think assertion functions should be capable of comparing the map object though 🤔

Reply:
I thought about flakiness too, but I opted to leave it like this until an actual problem arises.